components/eam/src/chemistry/yaml/cam_ndrop/f90_yaml/update_from_cldn_profile_beg_yml.f90

```diff
@@ -0,0 +1,172 @@
+        call open_files('update_from_cldn_profile', &  !intent-in
+        !     unit_input, unit_output) !intent-out
+            unit_input, unit_output, trim(ext_str)) !intent-out, with the use of ext_str
+
+
+        !start by adding an input string
+        !Use current timestep, given by y_nstep, to form name of input / output files
+        call write_input_output_header(unit_input, unit_output,yaml%lchnk_print,yaml%col_print, &
+             'update_from_cldn_profile',y_nstep, yaml%lev_print)
+
+        ! add code for writing data here
+        ! below are explicit input arguments to update_from_cldn_profile
+        
+        call write_var(unit_input,unit_output,'cldn_col_in',cldn_col_in(yaml%lev_print))
+        call write_var(unit_input,unit_output,'dtinv',dtinv)
+        call write_var(unit_input,unit_output,'wtke_col_in',wtke_col_in(yaml%lev_print))
+        call write_var(unit_input,unit_output,'zs',zs(yaml%lev_print))
+        call write_var(unit_input,unit_output,'dz',dz(yaml%lev_print))
+        call write_var(unit_input,unit_output,'temp_col_in',temp_col_in(yaml%lev_print))
+        call write_var(unit_input,unit_output,'csbot_cscen',csbot_cscen(yaml%lev_print))
+
+        call write_var(unit_input,unit_output,'raercol_nsav',raercol_nsav(yaml%lev_print,:))
+        call write_var(unit_input,unit_output,'raercol_cw_nsav',raercol_cw_nsav(yaml%lev_print,:))
+        call write_var(unit_input,unit_output,'nsource_col',nsource_col(yaml%lev_print))
+        call write_var(unit_input,unit_output,'qcld',qcld(yaml%lev_print))
+        call write_var(unit_input,unit_output,'factnum_col',factnum_col(yaml%lev_print,:))
+        call write_var(unit_input,unit_output,'ekd',ekd(yaml%lev_print))
+        call write_var(unit_input,unit_output,'nact',nact(yaml%lev_print,:))
+        call write_var(unit_input,unit_output,'mact',mact(yaml%lev_print,:))
+
+        ! the input value of state_q_col_in used is actually at level kp1 = min0(kk+1,pver),
+        ! where kk is the level of the output level-dependent fields.
+        ! for cs_col_in, values at both kk and kp1 are used for output at kk.
+        ! therefore, use these particular levels for the input fields, and indicate in 
+        ! the field name within the files
+        call write_var(unit_input,unit_output,'cs_col_in(kk)',cs_col_in(yaml%lev_print))
+        call write_var(unit_input,unit_output,'cs_col_in(kp1)',cs_col_in(kp1_yaml))
+        call write_var(unit_input,unit_output,'state_q_col_in(kp1)',state_q_col_in(kp1_yaml,:))
+
+        ! below are external module variable inputs to update_from_cldn_profile
+
+        call write_var(unit_input,unit_output,'top_lev',top_lev)
+        call write_var(unit_input,unit_output,'pver',pver)      
+
+        ! below are ndrop module variable inputs to update_from_cldn_profile
+
+
+        call write_var(unit_input,unit_output,'ntot_amode',ntot_amode)
+        call write_var(unit_input,unit_output,'nspec_amode',nspec_amode)
+
```

Comment:
Please add `cldn_col_in` and `raercol_nsav` at `kp1_yaml`.

Reply:
OK, these fields have been added.